plugins/thunderbolt/fu-thunderbolt-image.c

```diff
@@ -518,7 +518,7 @@ fu_plugin_thunderbolt_validate_image (GBytes  *controller_fw,
 		g_set_error (error,
 			     FWUPD_ERROR, FWUPD_ERROR_INVALID_FILE,
 			     "The FW image file is for a %s controller",
-			     is_host ? "device" : "host");
+			     is_host ? "host" : "device");
```

Comment:
`is_host` means that the controller is a host controller, so if the image isn't compatible it means that the image if for a device.

Reply:
@ybernat maybe I got just confused. Both test firmware blobs are for a host controller then?

Reply:
Yes. One is the image file as-is and the other is as it's read from the controller (after removal of the headers). I admit that I should have add a description. Sorry for that.

Reply:
Ah, now it makes sense. I guess we cant really test the device case then. Let me drop that commit and rename some variables ;)

Reply:
If you want one of them to look like it's (for) a device, change the second bit in location 0x10 or 0x11 (depends on which file you are changing). Currently it's 0x02, so the second bit is set, which means "a host controller". (see [line 491](https://github.com/hughsie/fwupd/pull/214/files/c65072211c75d31f6ca1b353313009b513df038f#diff-0d02b79a376b5c204ecf747b7877cbfaL491) and [line 498](https://github.com/hughsie/fwupd/pull/214/files/c65072211c75d31f6ca1b353313009b513df038f#diff-0d02b79a376b5c204ecf747b7877cbfaL498)

Reply:
You guys think it is worth generating that file on-the-fly for the test?

Reply:
Fuzzing? :)

Maybe just prepare a few interesting tests in advance (e.g. change location that matters, change location that shouldn't matter, change location that is masked in the bits inside/outside the mask) and save them as the data set.